luigi/scheduler.py

```diff
@@ -546,7 +546,8 @@ def add_task(self, worker, task_id, status=PENDING, runnable=True,
         * add additional workers/stakeholders
         * update priority when needed
         """
-        self.update(worker)
+        worker_id = kwargs['worker']
```

Comment:
This assumes 'worker' is always in kwargs (otherwise will raise KeyError). Maybe it's enforced somewhere (I really have no idea)? If not, the function comment should mention that this kwarg is required. Maybe even add an assert.